lib/tessel/deploy.js

```diff
@@ -342,854 +339,105 @@ actions.sendBundle = function(tessel, opts) {
-                  buffers.push(chunk);
-                })
-                .on('error', (data) => {
-                  reject(data);
-                })
-                .on('end', () => {
-                  fs.remove(tempBundleDir, (error) => {
-                    if (error) {
-                      reject(error);
-                    } else {
-                      resolve(Buffer.concat(buffers));
-                    }
-                  });
-                });
-            })
-            .catch(reject);
         }
-      });
-    });
-  } else {
 
-    return new Promise((resolve, reject) => {
-      // Copy the project to a temporary location.
-      // This allows us a safe way to "swap" binary modules.
-      fs.copySync(globRoot, tempBundleDir);
+        // When the stream closes, return from the function
+        remoteProcess.once('close', resolve);
 
-      return actions.injectBinaryModules(globRoot, tempBundleDir, opts)
-        .then(() => {
-          var fstream = new Ignore({
-            basename: '',
-            ignoreFiles: ['.tesselignore'],
-            path: tempBundleDir,
-          });
-
-          // Don't send the actual rules files
-          fstream.addIgnoreRules([
-            '**/.tesselignore',
-            '**/.tesselinclude',
-          ]);
-
-          if (includeNegateRules.length) {
-            fstream.addIgnoreRules(includeNegateRules);
-          }
-
-          if (!opts.single && includeFiles.length) {
-            // Instead of making a complete subclass of Ignore (as is done in fstream-npm,
-            // https://github.com/npm/fstream-npm/blob/master/fstream-npm.js#L91-L183),
-            // we'll over-ride the just the `applyIgnores` method for cases where there
-            // are .tesselinclude entries that have explicit inclusion rules.
-            fstream.applyIgnores = function(entry, partial, entryObj) {
-              if (includeFiles.indexOf(entry) !== -1) {
-                return true;
-              }
-
-              return Ignore.prototype.applyIgnores.call(fstream, entry, partial, entryObj);
-            };
-          }
-
-          if (opts.single) {
-            fstream.addIgnoreRules(['*', '!' + opts.resolvedEntryPoint]);
-          }
-
-          // This ensures that the remote root directory
-          // is the same level as the directory containing
-          // our program entry-point files.
-          fstream.on('entry', (entry) => {
-            entry.root = {
-              path: entry.path
-            };
-          });
-
-          // Send the ignore-filtered file stream into the tar packer
-          fstream.pipe(packer)
-            .on('data', (chunk) => {
-              buffers.push(chunk);
-            })
-            .on('error', (data) => {
-              reject(data);
-            })
-            .on('end', () => {
-              resolve(Buffer.concat(buffers));
-            });
-        })
-        .catch(reject);
-    });
-  }
-};
```

Comment:
Moved to `lib/tessel/deployment/javascript.js`. Now requires language specific implementation 

- [JavaScript](https://github.com/rwaldron/t2-cli/blob/deployment-reorg/lib/tessel/deployment/javascript.js#L371)
- [Python](https://github.com/rwaldron/t2-cli/blob/deployment-reorg/lib/tessel/deployment/python.js#L57)
- Rust not yet implemented